File: 1001-n-repeated-element-in-size-2n-array/1001-n-repeated-element-in-size-2n-array.cpp
class Solution {
public:
    int repeatedNTimes(vector<int>& nums) {
        int num_len = nums.size();
        vector<int> hash(10001, 0);
        int n = (num_len >> 1);
        for (int i = 0; i < num_len; ++i) {
            hash[nums[i]]++;
            if (hash[nums[i]] >= n)
                return nums[i];
        }
        return -1;
    }
};

File: 1001-n-repeated-element-in-size-2n-array/1001-n-repeated-element-in-size-2n-array.cs
public class Solution {
    public int RepeatedNTimes(int[] nums) {
        int[] hash = new int[10001];
        foreach (int num in nums) {
            hash[num]++;
            if (hash[num] >= (nums.Length >> 1))
                return num;
        }
        return -1;
    }
}